packages/cardano-graphql-services/src/DataProjection/blockHandlers/helpers.ts

```diff
@@ -0,0 +1,79 @@
+import { Asset, BigIntMath } from '@cardano-sdk/core';
+import { AssetMetadata } from '../../MetadataClient/types';
+import { Schema, isAlonzoBlock, isMaryBlock } from '@cardano-ogmios/client';
+
+export const getBlockType = (block: Schema.Block) => {
+  let b: Schema.BlockMary | Schema.BlockAlonzo | undefined;
+  if (isAlonzoBlock(block)) {
+    b = block.alonzo as Schema.BlockAlonzo;
+  } else if (isMaryBlock(block)) {
+    b = block.mary as Schema.BlockMary;
+  }
+  return b;
+};
+
+export const mapMetadata = (serverAssetMetadata: AssetMetadata): Asset.TokenMetadata => ({
+  decimals: serverAssetMetadata.decimals.value,
+  desc: serverAssetMetadata.description?.value,
+  icon: serverAssetMetadata.logo?.value,
+  name: serverAssetMetadata.name?.value,
+  ticker: serverAssetMetadata.ticker?.value,
+  url: serverAssetMetadata.url?.value
+  // Missing fields: sizedIcons
```

Comment:
This is fine for now. Could open an issue in metadata server repo. It should come as ["icon-\<size>"](https://github.com/cardano-foundation/CIPs/pull/137/files#diff-f478f61a4b67bc7fee1d41bbb427028a7615beaacf726efc9289104e64f26ffaR48). Let's keep the comment.